app/src/main/java/org/mozilla/focus/searchsuggestions/ui/SearchSuggestionsFragment.kt

```diff
@@ -116,7 +135,38 @@ class SearchSuggestionsFragment : Fragment(), CoroutineScope {
         }
 
         val searchSuggestionsView = view.findViewById<ComposeView>(R.id.search_suggestions_view)
-        searchSuggestionsView.setContent { SearchOverlay(searchSuggestionsViewModel) }
+
+        searchSuggestionsView.setContent {
+            val topSitesState =
+                requireComponents.appStore.observeAsComposableState { state -> state.topSites }
+
+            SearchOverlay(
+                viewModel = searchSuggestionsViewModel,
+                topSites = topSitesState.value!!,
+                topSitesMenuItems = listOfNotNull(
+                    TopSiteMenuItem(
+                        title = stringResource(R.string.remove_top_site),
+                        onClick = { topSite ->
+                            Shortcuts.shortcutRemovedCounter["removed_from_home_screen"].add()
+
+                            viewLifecycleOwner.lifecycleScope.launch(Dispatchers.IO) {
+                                requireComponents.topSitesUseCases.removeTopSites(topSite)
+                            }
+                        }
+                    )
+                ),
+                onTopSiteClick = { topSite ->
+                    Shortcuts.shortcutOpenedCounter.add()
+
+                    requireComponents.tabsUseCases.addTab(
+                        url = topSite.url,
+                        source = SessionState.Source.Internal.HomeScreen,
+                        selectTab = true,
+                        private = true
+                    )
+                }
```

Comment:
This is more or less copied over from UrlInputFragment https://searchfox.org/mozilla-mobile/source/focus-android/app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.kt#303-324.

I am wondering if there is a recommended way to extract this out so that it can be reusable in both fragments.